File: 0645-set-mismatch/0645-set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        unordered_map<int, int> mp;
        for (int& n : nums) {
            mp[n]++;
        }
        vector<int> res(2);
        for (int i = 1; i <= nums.size(); i++) {
            if (mp[i] == 0) {
                res[1] = i;
            }
            if (mp[i] == 2) {
                res[0] = i;
            }
        }
        return res;
    }
};

File: 0645-set-mismatch/0645-set-mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            nums[(nums[i] - 1) % n] += n;
        }
        int[] result = new int[2];
        for (int i = 0; i < n; i++) {
            if (nums[i] > 2 * n) {
                result[0] = i + 1;
            }
            if (nums[i] <= n) {
                result[1] = i + 1;
            }
        }
        return result;
    }
}